client/state/terms/selectors.js

```diff
@@ -169,3 +168,17 @@ export function getTerms( state, siteId, taxonomy ) {
 
 	return manager.getItems();
 }
+
+/**
+ * Returns a term object by site ID, taxonomy, and termId
+ *
+ * @param  {Object}  state    Global state tree
+ * @param  {Number}  siteId   Site ID
+ * @param  {String}  taxonomy Taxonomy slug
+ * @param  {Number}  termId   Term ID
+ * @return {?Object}          Term object
+ */
+export const getTerm = createSelector(
+	( state, siteId, taxonomy, termId ) => find( getTerms( state, siteId, taxonomy ), { ID: termId } ) || null,
```

Comment:
[`TermQueryManager#getItem`](https://github.com/Automattic/wp-calypso/blob/fa86649bfcab68568ea0e88a49c06319614d7e54/client/lib/query-manager/index.js#L133-L141) will be more efficient here.